### PR TITLE
Remove the OcTextInput component from the OcSearchBar component

### PR DESCRIPTION
**Pull request by JammingBen**
Also replace the loading spinner with the OcSpinner component in the course of this.

Related to https://github.com/owncloud/owncloud-design-system/pull/1141 as this PR removes placeholders from the `OcTextInput` component.

### CODE REVIEW

**Comment by JammingBen** [comment]
I still need to investigate on "How-To-Aria" for the loading spinner.

**Comment by kulmann** [comment]
Screenshot 2021-03-16 at 13 35 31

The border radius is visible when a button exists. That was not the case before.

File: src/components/OcSearchBar.vue [hunk 1]
```diff
@@ -8,21 +8,15 @@
...
       <span v-if="icon" class="uk-form-icon">
         <oc-icon v-show="!loading" :name="icon" />
-        <div
-          v-show="loading"
-          :class="$_ocSpinner_class"
-          :aria-label="ariaLabel"
-          tabindex="-1"
-          role="img"
-        />
+        <oc-spinner v-show="loading" :size="spinnerSize" aria-label="Loading results" />
```
Comment (kulmann) [inline src/components/OcSearchBar.vue:11]: I know that the aria-label was entirely broken before. Could you add a required property `loadingAccessibleLabel` (or similar naming) for this? Needs to be provided by the code that uses the searchbar.

**Comment by JammingBen** [comment]
> The border radius is visible when a button exists. That was not the case before.

Fixed 👍 

Now the only thing that differs visually from the original implementation is the margin right to the small loading spinner. This happens due to the use of the `OcSpinner` component. I find it quite nice though, so I would just leave it that way?

**Review by kulmann — APPROVED** [review]
Nice 👍

File: src/components/OcSearchBar.vue [hunk 2]
```diff
@@ -149,6 +143,13 @@ export default {
       required: false,
       default: false,
     },
+    /**
+     * The aria label for the loading spinner
+     */
+    loadingAccessibleLabel: {
+      type: String,
+      required: true,
```
Comment (kulmann) [inline src/components/OcSearchBar.vue:151]: Thinking about it, this is a breaking change. We have `$gettext` available. Could you check if you can set a sensible default in a callback?

**Review by kulmann — CHANGES_REQUESTED** [review]

File: src/components/OcSearchBar.vue (hunk 2 above)
Comment (JammingBen) [reply src/components/OcSearchBar.vue]: Please check again. `$gettext` cannot be used as default value (at least I didn't find a way to do so), thus I had to implement a computed property in addition.

Comment (kulmann) [reply src/components/OcSearchBar.vue]: Ok, that's fine. But please use `loadingAccessibleLabel` for the prop and define another name for the computed property. The props are the public API of the component and should have good names. The names for the computed properties are not that important (of course still for us as developers of the design system ;-) ).

File: src/components/OcSearchBar.vue [hunk 3]
```diff
@@ -158,20 +162,27 @@ export default {
...
+    inputClass() {
       const classes = ["oc-search-input"]
 
       this.icon && classes.push("oc-search-input-icon")
       !this.buttonHidden && classes.push("oc-search-input-button")
 
       return classes
     },
+    loadingAccessibleLabel() {
+      if (this.customLoadingAccessibleLabel) {
```
Comment (kulmann) [inline src/components/OcSearchBar.vue:180]: Can be simplified to `this.customLoadingAccessibleLabel || this.$gettext("Loading results")`.

**Review by kulmann — APPROVED** [review]
Nice 🚀

**Pull request merged**